apps/gdalalg_raster_tile.cpp

```diff
@@ -3668,9 +3684,11 @@ bool GDALRasterTileAlgorithm::RunImpl(GDALProgressFunc pfnProgress,
         // do not generate base tiles if called as a child process with
         // --ovr-zoom-level
     }
-    else if (m_numThreads > 1 &&
+    else if (m_numThreads > 1 && nBaseTiles > 1 &&
              ((m_parallelMethod.empty() &&
-               nBaseTiles / m_numThreads > THRESHOLD_TILES_PER_JOB &&
+               m_numThreads >= THRESHOLD_MIN_THREADS_FOR_SPAWN &&
+               nBaseTiles >=
+                   THRESHOLD_MIN_THREADS_FOR_SPAWN * THRESHOLD_TILES_PER_JOB &&
```

Comment:
This has the same problem as before. Think of 8 as MAX_THREADS_FOR_THREAD and never launch more than that number of threads without switching to spawn.

Reply:
not totally sure to follow you. I've added a commit that limit the maximum number of threads (for the threading use case) to min(number_of_threads_specified_by_user_or_all_available, number_of_tiles / 100) . If that's not what you've in mind, a diff / pull request with your proposed changes could make it easier

Reply:
Now that you have limited m_numThreads you can just eliminate the "nBaseTiles >= ..." condition and it's good.